prov/gni/src/gnix_msg.c

```diff
@@ -456,6 +439,29 @@ static int __recv_completion_src(
 	return FI_SUCCESS;
 }
 
+/*
+ * GNI provider generates a special CQE to indicate
+ * its releasing a FI_MULTI_RECV receive buffer back to
+ * the application.
+ */
```

Comment:
this is one of the main things about this PR.  We have an out-of-order network so we may be filling different parts of the multi-receive out of order wrt to where they started in the buffer.  We do not know which transfer will be completed last if we are using rendezvous for some of the messages.  So hence the use of a special CQE to indicate that the buffer is 1) full and 2) all the outstanding message data from the various messages arriving in the buffer have been delivered.  The original design would only work correctly if all the messages coming in to the buffer were using the eager, SMSG path.